### PR TITLE
ROX-25786: Create FilteredWorkflowSelector component

### PR DESCRIPTION
**Pull request by sachaudh**
### Description

This PR adds a new selector to the Policy Violations to choose a filtered workflow (`Application view`, `Platform view`, `Full view`). The state for the filtered workflow will be kept in the URL as `filteredWorkflowState`. The value will be used to filter the table in Policy Violations. That functionality will come in the following PR once the backend adds the capability. We will disable the `Application view` and `Platform view` for now.

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [x] CI results are inspected

#### Automated testing

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [x] modified existing tests
- [] no added tests

#### How I validated my change

Screenshot 2024-10-01 at 9 29 04 PM
Screenshot 2024-10-01 at 9 29 10 PM
Screenshot 2024-10-01 at 9 29 19 PM

### CODE REVIEW

File: ui/apps/platform/src/Components/FilteredWorkflowSelector/FilteredWorkflowSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    MenuToggle,
+    Select,
+    SelectList,
+    SelectOption,
+} from '@patternfly/react-core';
+import { CubesIcon } from '@patternfly/react-icons';
+
+import { FilteredWorkflowState, SetFilteredWorkflowState } from './types';
+
+const width = '330px';
```
Comment (pedrottimark) [inline ui/apps/platform/src/Components/FilteredWorkflowSelector/FilteredWorkflowSelector.tsx:15]: As a warm up to future evaluation of accessibility, can you verify that width of `Select` element is okay if you zoom in? If not, is `em` or `rem` an alternative?

> Except for captions and images of text, text can be resized without assistive technology up to 200 percent without loss of content or functionality.

https://www.w3.org/TR/WCAG21/#resize-text

File: ui/apps/platform/src/Containers/Violations/ViolationsTablePage.tsx [hunk 2]
```diff
@@ -28,6 +29,8 @@ import useURLSearch from 'hooks/useURLSearch';
 import useURLPagination from 'hooks/useURLPagination';
 import useInterval from 'hooks/useInterval';
 import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+import FilteredWorkflowSelector from 'Components/FilteredWorkflowSelector/FilteredWorkflowSelector';
+import useFilteredWorkflowState from 'Components/FilteredWorkflowSelector/useFilteredWorkflowState';
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Violations/ViolationsTablePage.tsx:33]: Super, this is example of what the team discussed recently, that cause and effect relationship is clearest if URL hooks are at same level.

One question about naming where my memory fails even after reviewing similar recent work: do hooks that affect URL also have URL in their names?

**Review by pedrottimark — APPROVED** [review]
A request and a question for your consideration.

File: ui/apps/platform/src/Components/FilteredWorkflowSelector/FilteredWorkflowSelector.tsx (hunk 1 above)
Comment (sachaudh) [reply ui/apps/platform/src/Components/FilteredWorkflowSelector/FilteredWorkflowSelector.tsx:15]: It looks good to me. I kept the same pattern as we see in Vuln Management - Workload CVEs.

File: ui/apps/platform/src/Containers/Violations/ViolationsTablePage.tsx (hunk 2 above)
Comment (sachaudh) [reply ui/apps/platform/src/Containers/Violations/ViolationsTablePage.tsx:33]: Yea, we can make it easier to discern that the hook affects the URL state by having the URL in the name. That's a good idea. If we don't do that already, I think its a good pattern.

**Pull request merged**